filesystems.py

```diff
@@ -0,0 +1,18 @@
+import tempfile
+from typing import TextIO
+
+from pyftpdlib.filesystems import AbstractedFS
+
+
+class OpenSPPFS(AbstractedFS):
+    def open(self, filename: str, mode: str) -> TextIO:
+        dir_tree = filename.split("/")
+        new_filename = "_".join(dir_tree)
+        # Use mkdtemp, so it persist and should be removed on
+        # OpenSPPFTPHandler.on_file_received after uploading to OpenSPP
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Can we use in-memory file?

Reply:
Tried this last time but it is being closed somewhere before we can use it for upload. So when we try to catch the file in `OpenSPPFTPHandler.on_file_received`, it is already deleted. I tried finding the path where it is closed but got stucked in `FTPHandler.ftp_STOR`. I skipped the approach and use tempfile instead to catch up with the demo. Do you want me to revisit this approach?

Reply:
Added changes for in-memory file